tensorflow_quantum/core/src/program_resolution.h

```diff
@@ -42,12 +42,14 @@ tensorflow::Status ResolveQubitIds(
 // Resolves all of the symbols present in the Program. Iterates through all
 // operations in all moments, and if any Args have a symbol, replaces the one-of
 // with an ArgValue representing the value in the parameter map keyed by the
-// symbol. Returns an error if the parameter value cannot be found.
-// TODO(pmassey): Consider returning an error if a value in the parameter map
-// isn't used.
+// symbol. When `resolve_all` is true, returns an error if a symbol does not
+// have a correponding value in `param_map`.  When `flag_unused` is true,
+// returns an error if a value in the parameter map isn't used.
+// TODO(zaqqwerty): Work to flip default of `flag_unused` to true.
```

Comment:
In adding `flag_unused` we are making room for new error mechanisms that are not required by this PR and are not required currently by the rest of TFQ. What's the motivation for adding them here and now ?

Reply:
Left over from some other attempts at things, reverted it now.